src/ripple/app/consensus/RCLValidations.cpp

```diff
@@ -240,24 +240,45 @@ RCLValidations::onStale(RCLValidation&& v)
         return;
 
     writing_ = true;
-    app_.getJobQueue().addJob(
-        jtWRITE, "Validations::doWrite", [this](Job&) { doWrite(); });
+    app_.getJobQueue().addJob(jtWRITE, "Validations::doWrite", [this](Job&) {
+
+        auto event =
+            app_.getJobQueue().getLoadEventAP(jtDISK, "ValidationWrite");
+        ScopedLockType sl(lock_);
+        doWrite(sl);
+    });
 }
 
 void
 RCLValidations::flush()
 {
     JLOG(j_.info()) << "Flushing validations";
-    ScopedLockType sl(lock_);
+    bool anyNew = false;
+    {
+        ScopedLockType sl(lock_);
 
-    Validations::flush();
+        Validations::flush([&](RCLValidation && v)
```

Comment:
If I move the lock inside, do I need to consider concurrent calls to `RCLValidations::flush`?  My understanding is that the lock protects _all_ data members of the class, both in the old and new code.  If another call to `RCLValidations::flush` occurs, wouldn't this cause a data race?

As long as I document the semantics of `flush`, I'm not too concerned about calling it while holding the lock and would prefer to avoid thrashing.  If we are comfortable calling the external code in `.emplace_back` while holding a lock, I think extending that comfort to the `flush` function is acceptable.   Does that seem reasonable or would you prefer I move the lock inside?